projects/RabbitMQ.Client/client/api/IConnection.cs

```diff
@@ -234,8 +234,17 @@ Task CloseAsync(ushort reasonCode, string reasonText, TimeSpan timeout, bool abo
         /// <summary>
         /// Asynchronously create and return a fresh channel, session, and channel.
         /// </summary>
+        /// <param name="dispatchConsumerConcurrency">
+        /// Set to a value greater than one to enable concurrent processing. For a concurrency greater than one <see cref="IAsyncBasicConsumer"/>
+        /// will be offloaded to the worker thread pool so it is important to choose the value for the concurrency wisely to avoid thread pool overloading.
+        /// <see cref="IAsyncBasicConsumer"/> can handle concurrency much more efficiently due to the non-blocking nature of the consumer.
+        ///
+        /// Defaults to <see cref="IConnectionFactory.ConsumerDispatchConcurrency"/>.
```

Comment:
I wonder if `IConnectionFactory.ConsumerDispatchConcurrency` should be called now `IConnectionFactory.DefaultConsumerDispatchConcurrency`

Reply:
I thought the whole ``ConsumerDispatchConcurrency `` is basically legacy logic for blocking/sync consumers? At least that's how I understand the comment (last line).
Do you really want to "polute" the interface with that, so everyone has to either pass null in there or do ``cancellationToken: cancellationToken`` for named arguments?

Reply:
> I thought the whole ConsumerDispatchConcurrency is basically legacy logic for blocking/sync consumers? At least that's how I understand the comment (last line).

While I see your point with your interpretation of the comment this is not actually what happens under the cover. 

The consumer dispatcher concurrency influences how the `ConsumerDispatcherChannelBase` behaves. By default, one reader reads the channel and dispatches the async consumer and awaits the completion. That means messages are handled one by one.

By setting it to a higher concurrency value, multiple reader loops are spawned to the level of concurrency, effectively allowing concurrent processing of messages arriving at the channel. 

> Do you really want to "polute" the interface with that, so everyone has to either pass null in there or do cancellationToken: cancellationToken for named arguments?

FYI Quite a few methods on the client require explicitly specifying `cancellationToken: ` due to the number of optional overloads. So that pattern is nothing new and adding one more doesn't spill the glass.

Reply:
Ah thank you for the explanation, that actually explains why I had to build my own dispatcher via Channels so the messages were processed in parallel.